go/porcelain/deploy.go

```diff
@@ -83,7 +83,8 @@ type DeployOptions struct {
 	BuildDir          string
 	LargeMediaEnabled bool
 
-	IsDraft bool
+	IsDraft   bool
+	SkipRetry bool
```

Comment:
Is the idea that buildbot will set this property based on the value of a feature flag? If so, I imagine we'll remove this once we finish the rollout of the flag?

This only controls the retry mechanism for functions, not regular files, so it might be worth to name the variable accordingly, or at the very least add a comment explaining what it's for.

Reply:
Is it worth using this feature flag for both function _and_ file uploads, since we don't want to retry 400/422 file uploads either? Here's the issue for that: https://github.com/netlify/open-api/issues/476

As written currently, this does change the behavior for file uploads. Erik mentioned it in [a comment in his original draft](https://github.com/netlify/open-api/pull/478/commits/2f75b17dec49ef7d09c3126c28d69268ddba52f9#diff-602e1ebf353d4906b9de529630317e7f88f5832aa50bbb1ca269741412e1c112R548). 

If we want to keep the scope of this flag/pr to _just_ handle function uploads for now, that's totally fine too, we would just change [this line ](https://github.com/netlify/open-api/pull/478/files#diff-602e1ebf353d4906b9de529630317e7f88f5832aa50bbb1ca269741412e1c112R350) from:
```
n.uploadFiles(ctx, deploy, options.files, options.Observer, fileUpload, options.UploadTimeout, skipRetry)
```
to 
```
n.uploadFiles(ctx, deploy, options.files, options.Observer, fileUpload, options.UploadTimeout, false)
```

Reply:
Ah, my bad! I was reviewing this on my phone last night and the diff made it seem that the check for `skipRetry` was under the `case functionUpload:` branch. Ignore me.

If stopping retries for files is also something we want to do, then that makes sense!